client/src/pages/settings/settings.page.js

```diff
@@ -22,17 +22,173 @@ import FormControlLabel from "@mui/material/FormControlLabel";
+        }
+        /* INCLUDE ACTUAL ERROR HANDLING HERE/DISPLAYING THE MESSAGE */
+        const salt = await bcrypt.genSalt(10);
+        const hash = await bcrypt.hash(data.get("new-password"), salt);
+        let relevant_data = JSON.stringify({
+            password: hash,
+        });
+        try {
+            // Send data to the backend via PUT
+            fetch(`http://localhost:8080/users/${id}`, {
+                method: "PUT",
+                headers: { "Content-Type": "application/json" },
+                body: relevant_data, // body data type must match "Content-Type" header
+            })
+                .then((response) => response.json())
+                .then((data) => {
+                    console.log(data);
+                })
+                .catch((error) => console.error(error));
+            //TODO ADD A REFRESH HERE?
+        } catch (err) {
+            console.error(err);
+        }
+    };
+
+    const handleAdvancedSubmit = (event) => {
+        event.preventDefault();
+        const data = new FormData(event.currentTarget);
+
+        let json_data = JSON.stringify({
+            location: data.get("location"),
+            birthday: data.get("birthday"),
+            phoneNumber: data.get("phone-number"),
+            imgUrl: data.get("imgUrl"),
+            bio: data.get("bio"),
+            twitterHandle: data.get("twitterHandle"),
+        });
+        console.log(json_data);
+        const id = auth._id;
+        console.log(id);
+        try {
+            // Send data to the backend via PUT
+            fetch(`http://localhost:8080/users/${id}`, {
+                method: "PUT",
+                headers: { "Content-Type": "application/json" },
+                body: json_data, // body data type must match "Content-Type" header
+            })
+                .then((response) => response.json())
+                .then((data) => {
+                    console.log(data);
+                })
+                .catch((error) => console.error(error));
+        } catch (err) {
+            console.error(err);
+        }
+    };
+
+    const handleDeleteAccount = (event) => {
+        event.preventDefault();
+        console.log("Are you sure?");
+        const id = auth._id;
+        console.log(id);
+        try {
+            // Send data to the backend via PUT
+            fetch(`http://localhost:8080/users/${id}`, {
+                method: "DELETE",
+                headers: { "Content-Type": "application/json" },
+            })
+                .then((response) => response.json())
+                .then((data) => {
+                    console.log(data);
+                    console.log("DELETED");
+                    //TODO REDIRECT TO HOMEPAGE
+                })
+                .catch((error) => console.error(error));
+        } catch (err) {
+            console.error(err);
+        }
+    };
+
+    //TODO TEST AFTER RECIPES ARE PROPERLY ADDED
+    const handleResetAccount = (event) => {
```

Comment:
The way you have written this right now will just delete all recipes from account and I don't know if that's clear from front end. It may be better to make this erase all settings other than basic info. There should also be a check to make sure they want to do this (same with delete account). It may make sense to move then functionality that exists as is to the my recipes page.